.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,153 @@
+
+### Requesting a new feature
+
+If there's a new feature that you want to see added to CKAN, you can [open an issue][2] using the respective 'Feature' template, or you can write the code yourself.
+
+## Helping with issues
+### Verifying issues
+
+For starters, it helps just to verify bug reports. Can you reproduce the reported issue on your own computer? If so, you can add a comment to the issue saying that you're seeing the same thing.
+
+If an issue is very vague, can you help narrow it down to something more specific? Maybe you can provide additional information to help reproduce a bug, or help by eliminating needless steps that aren't required to demonstrate the problem.
+
+### Testing patches
+
+You can also help out by examining pull requests that have been submitted to CKAN via GitHub. In order to apply someone's changes, you need to first create a dedicated branch:
+
+```
+$ git clone https://github.com/KSP-CKAN/CKAN.git
+$ cd CKAN
+$ git checkout -b testing_branch
+```
+
+Then, you can use their remote branch to update your codebase. For example, let's say the GitHub user JohnSmith has forked and pushed to a topic branch "orange" located at https://github.com/JohnSmith/CKAN.
+
+```
+$ git remote add JohnSmith https://github.com/JohnSmith/CKAN.git
+$ git fetch JohnSmith
+$ git checkout JohnSmith/orange
+```
+
+After applying their branch, test it out! Comment on the GitHub issue to apply some changes or by giving your approval.
+
+## Translating
+
+We recently started working on translating CKAN and are happy to have people volunteer to translate with us. Just follow these steps:
+
+* Fork this repository by clicking on the 'Fork' button on the top of this page.
+
+* Click on the green 'Clone' button and copy the link.
+
+* Open a terminal and run following code:
+
+```
+$ git clone https://github.com/YourName/CKAN.git
+$ cd CKAN
+$ git checkout -b new_branch
+``` 
+
+* Add a folder for your own language, for example: ` GUI/Localization/it-IT` for Italian.
+
+* Copy the contents of ` GUI/Localization/en-US` into your own language folder.
+
+* Rename the files from `en-US` to `it-IT`.
+
+* Start translating the files.
+
+**Translations we already have:**
+
+* English (US)
+* English (AU)
+* German
+
+## Creating pull requests
+
+Before you make a pull request, you have to agree to contribute under these licenses:
+
+* Contributions to the CKAN source *must* be licensed under the [MIT license][3].
+
+* Contributions of CKAN metadata files (.netkan or .ckan) *must* be be contributed under the [CC-0][4] license.
+
+### Creating a patch
+
+* Fork this repository by clicking on the 'Fork' button on the top of this page.
+
+* Click on the green 'Clone' button and copy the link.
+
+* Open a terminal and run following code:
+
+```
+$ git clone https://github.com/YourName/CKAN.git
```

Comment:
It's really convenient to be able to use `git checkout master; git pull` to keep my local master up to date. And it might be necessary if we need to ask somebody to rebase a pull request after a conflicting PR gets merged. Is that possible if they clone their own fork?

Reply:
What is the best way to rebase? I use a alias for it: `git rebase -i 'git merge-base master HEAD'`
Or do it this way:
```
$ git fetch
$ git checkout new_branch
$ git rebase KSP-CKAN/master  <- Is this correct?
```

Reply:
I guess there are multiple ways; what I do is:

```sh
git checkout master
git pull
git checkout fix/whatever
git rebase master
```

... since I think that's how various git tutorials explained it. But this depends on master's upstream being KSP-CKAN.

Can we add this after the clone?

```sh
git remote add upstream https://github.com/KSP-CKAN/CKAN.git
```

Then if they need to rebase:

```sh
git fetch --all
git rebase upstream/master
```